src/classes/LVL_Level_TDTM.cls

```diff
@@ -0,0 +1,166 @@
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2016
+* @group Levels
+* @group-content ../../ApexDocContent/Levels.htm
+* @description Trigger handler on Level object, ensures Levels within the same ladder do not overlap.
+*/
+public class LVL_Level_TDTM extends TDTM_Runnable {
+
+    /*******************************************************************************************************
+    * @description Trigger Handler on Level, which validates Levels within the same ladder do not overlap.
+    * @param listNew the list of Levels from trigger new. 
+    * @param listOld the list of Levels from trigger old. 
+    * @param triggerAction Which action triggered the current context. (Before or After; Insert, Update, 
+    * Delete, or Undelete.)
+    * @param objResult The field describe object for the current trigger context. 
+    * @return dmlWrapper Objects that need to be inserted or updated after the processing of the trigger.
+    ********************************************************************************************************/
+    public override DmlWrapper run(List<SObject> newlist, List<SObject> oldlist, 
+        TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
+            
+            // only validate before triggers
+            if (triggerAction == TDTM_Runnable.Action.BeforeInsert || triggerAction == TDTM_Runnable.Action.BeforeUpdate) {
+                
+                // first get a set of all objects being referenced
+                set<String> setStrObject = new set<String>();
+                for (SObject sobj : newlist) {
+                    setStrObject.add(((Level__c)sobj).Target__c);
+                }
+                
+                // for each object, validate any updates to its ladders
+                for (String strObj : setStrObject) {
+                    
+                    // get our map of Ladder Levels
+                    map<String, list<Level__c>> mapLadderToLevels = LVL_LevelAssign_BATCH.mapLadderToLevelsForObject(strObj);
+                    
+                    // go thru our new/updated levels
+                    for (integer i = 0; i < newlist.size(); i++) {
```

Comment:
Here we're looping through the entire trigger set, even though we've only got the mapLadderToLevels for one sObject type at this point. I think in this for loop you need to make sure we're only dealing with levels of the correct sObject type. Also we're running a query for each sObject, it would be preferable to run a single query for all sObject types in the trigger set, although it's not a big deal in this transaction.